src/Microsoft.Diagnostics.Monitoring.HostingStartup/ParameterCapturing/Pipeline/ParameterCapturingPipeline.cs

```diff
@@ -154,13 +156,36 @@ public bool TryComplete()
             return _requestQueue.Writer.TryComplete();
         }
 
+        private void ValidateMethods(MethodDescription[] methods)
+        {
+            List<MethodDescription> _deniedMethodDescriptions = new();
+            foreach (MethodDescription methodDescription in methods)
+            {
+                foreach (string deniedType in _typeDenyList)
+                {
+                    if (TypeUtils.IsSubType(deniedType, methodDescription.TypeName))
+                    {
+                        _deniedMethodDescriptions.Add(methodDescription);
+                        break;
+                    }
+                }
+            }
+
+            if (_deniedMethodDescriptions.Count > 0)
+            {
+                throw new DeniedMethodsExceptions(_deniedMethodDescriptions);
+            }
+        }
+
         public void SubmitRequest(StartCapturingParametersPayload payload)
         {
             if (payload.Methods.Length == 0)
             {
                 throw new ArgumentException(nameof(payload.Methods));
             }
 
+            ValidateMethods(payload.Methods);
```

Comment:
Could this be made into a singleton service? This would alleviate the need for the pipeline to understand this policy and focus on technical execution and would allow easier direct testing of the actual behavior (e.g. the service implementation is tested rather than just the helper method).